tests/golos.publication_tests.cpp

```diff
@@ -194,12 +215,23 @@ BOOST_FIXTURE_TEST_CASE(create_message, golos_publication_tester) try {
     BOOST_TEST_MESSAGE("Create message testing.");
     init();
     init_params();
+    init_referral();
+
     BOOST_CHECK_EQUAL(success(), post.create_msg(N(brucelee), "permlink"));
 
+    auto expire = 8; // sec
+    BOOST_CHECK_EQUAL(success(), referral.create_referral(N(dan.larimer), N(chucknorris), 500,
+                                                      cur_time().to_seconds() + expire,
+                                                      asset(50000, _sym)));
+
     BOOST_TEST_MESSAGE("--- checking that another user can create a message with the same permlink.");
     BOOST_CHECK_EQUAL(success(), post.create_msg(N(chucknorris), "permlink"));
 
     auto id = hash64("permlink");
+    auto post_chucknorris = post.get_message(N(chucknorris), id);
+    auto size_ben = post_chucknorris["beneficiaries"].size();
+    BOOST_CHECK_EQUAL( post_chucknorris["beneficiaries"][size_ben - 1].as<structures::beneficiaries>().account, N(dan.larimer) );
```

Comment:
This check is not enough. We need to check that he has the right percentage of reward. Still need to make a check for incorrect input.